src/ApacheOrcDotNet.OptimizedReader/Infrastructure/MemoryMappedFileRangeProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Threading.Tasks;
+
+namespace ApacheOrcDotNet.OptimizedReader.Infrastructure
+{
+    public sealed class MemoryMappedFileRangeProvider : IByteRangeProvider
+    {
+        readonly MemoryMappedFile _memoryMappedFile;
+        readonly long _length;
+
+        internal MemoryMappedFileRangeProvider(string path)
+        {
+            using (var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read))
+                _length = fileStream.Length;
```

Comment:
It was to be sure that no open files would remain, but I had a wrong idea about the `FileInfo` class. Always thought it would hold a handle to the file.